### PR TITLE
opengl and serialization test updates

### PR DESCRIPTION
**Pull request by headupinclouds**
opengl related acf fixes and …
* support chnsCompute + isInit: see #37
* support rgb-to-gray w/ correct channel count==1 : see #36
* `*.mat` serialization testing
  + download Inria/Caltech *.mat toolbox pedestrian models from github storage using cmake/hunter_private_data
  + add simple serialization tests in test-acf GTest code (more rigorous testing needed)
  + remove unused code blocks and some debug code in test-acf.cpp
* fix a boundary case in sse gradHist() code
* remove unused conditional cv::cvtColor() code blocks in rgbConvert.cpp
* add virtual destructor to ObjectDetector.h base class
* support acf shrink (via constructor input) in the GPUACF code
* update gradhist glsl shader to highp precision
* unify header include style
* remove various comment blocks + format fixing
* add acf special member isLuv to indicate special case input LUV processing (such as in gpgpu pipeline)
* provide naive GLDetector : public acf::Detector naive detector to exercise shader code in acf-detect
  + use aglet lib for portable context
  + compute acf pyramids from input cv::Mat (in practice this would be separated for efficiency

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -186,6 +186,38 @@ endif(ACF_SERIALIZE_WITH_CEREAL)
 
 message("ACF_ADD_TO_STRING : ${ACF_ADD_TO_STRING}")
 
+################
+## TEST DATA ###
+################
+
+if(COMMAND hunter_private_data)
```
[the remark this answers is not in the document]
Comment (headupinclouds) [reply CMakeLists.txt]: We don't need it.  I've removed it.  Thanks.

File: CMakeLists.txt [hunk 2]
```diff
@@ -186,6 +186,47 @@ endif(ACF_SERIALIZE_WITH_CEREAL)
 
 message("ACF_ADD_TO_STRING : ${ACF_ADD_TO_STRING}")
 
+################
+## TEST DATA ###
+################
+
+# Only download test data if tests are enabled and we are running 
+# a recent enough version of Hunter to support hunter_private_data()
+if(ACF_BUILD_TESTS AND COMMAND hunter_private_data)
```
Comment (ruslo) [inline CMakeLists.txt]: `if(... AND COMMAND hunter_private_data)` would it be better to return an error and ask for update?

Comment (headupinclouds) [reply CMakeLists.txt]: I've added a `FATAL_ERROR` in this case with a message telling the user to update Hunter.
